src/main/java/data_access/InMemoryUserDataAccessObject.java

```diff
@@ -22,7 +22,7 @@ public User getUserWithCredential(String credential) throws ValidationException
         return getUserWithUsername(username);
     }
 
-    private User getUserWithUsername(String username) {
+    public User getUserWithUsername(String username) {
```

Comment:
This shouldn't be public. This is used in getUserWithCredential to select the current user from the in memory database.